packages/application-shell/src/components/navbar/use-navbar-state-manager.ts

```diff
@@ -71,6 +71,14 @@ const reducer = (state: State, action: Action): State => {
   }
 };
 
+const isFocusOutEventCalledBySubmenuItem = (event: FocusEvent) =>
+  // a case when a submenu item loses focus
+  event.type === 'focusout' &&
+  // element receiving focus
+  (event.relatedTarget as Element)?.matches('a[class*="__text-link__"]') &&
+  // element losing focus
+  (event.target as Element)?.matches('a[class*="__text-link-sublist__"]');
```

Comment:
I would suggest to use more "stable" selectors. We shouldn't rely on class names, as they might change depending on how we style things.

Can we use data attributes for example?

Reply:
Thank you! Updated in 7b31dfb06b4db1c6c623a7b8c9589bcf3cb9cbcb. 
I will create a task for the follow-up work.